src/infra/sslQueryRepo.go

```diff
@@ -0,0 +1,223 @@
+package infra
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+
+	"github.com/speedianet/sam/src/domain/entity"
+	"github.com/speedianet/sam/src/domain/valueObject"
+	infraHelper "github.com/speedianet/sam/src/infra/helper"
+)
+
+type SslQueryRepo struct {
+	olsHttpdConfigPath string
+}
+
+type HttpdVhostConfig struct {
+	VirtualHost string
+	FilePath    string
+}
+
+type VhostConfig struct {
+	FilePath    string
+	FileContent string
+}
+
+func NewSslQueryRepo() *SslQueryRepo {
+	return &SslQueryRepo{
+		olsHttpdConfigPath: "/usr/local/lsws/conf/httpd_config.conf",
+	}
+}
+
+func (repo SslQueryRepo) splitSslCertificate(
+	sslCertContent string,
+) ([]entity.SslCertificate, error) {
+	certificates := []entity.SslCertificate{}
+
+	sslCertContentSlice := strings.SplitAfter(sslCertContent, "-----END CERTIFICATE-----\n")
+	for _, sslCertContentStr := range sslCertContentSlice {
+		certificate, err := entity.NewSslCertificate(sslCertContentStr)
+		if err != nil {
+			return certificates, errors.New("SslCertificateError")
+		}
+		certificates = append(certificates, certificate)
+	}
+
+	return certificates, nil
+}
+
+func (repo SslQueryRepo) SslFactory(
+	sslHostname string,
+	sslPrivateKey string,
+	sslCertContent string,
+) (entity.SslPair, error) {
+	var ssl entity.SslPair
+
+	hostname, err := valueObject.NewFqdn(sslHostname)
+	if err != nil {
+		return ssl, errors.New("SslHostnameError")
+	}
+	_, err = repo.GetVhostConfig(hostname.String())
+	if err != nil {
+		return ssl, err
+	}
+
+	privateKey, err := entity.NewSslPrivateKey(sslPrivateKey)
+	if err != nil {
+		return ssl, err
+	}
+
+	sslCertificates, err := repo.splitSslCertificate(sslCertContent)
+	if err != nil {
+		return ssl, err
+	}
+
+	var certificate entity.SslCertificate
+	var chainCertificates []entity.SslCertificate
+	for _, sslCertificate := range sslCertificates {
+		if sslCertificate.IsCA {
+			certificate = sslCertificate
+			continue
+		}
+
+		chainCertificates = append(chainCertificates, sslCertificate)
+	}
+
+	id, err := valueObject.NewSslSerialNumber(certificate.SerialNumber.String())
+	if err != nil {
+		return ssl, err
+	}
```

Comment:
O SerialNumber do certificate já não é um VO do SslSerialNumber?

Reply:
Pensei em fazer um _double check_ aqui, mas olha por fora parece inútil. Vou ajustar!